Makefile

```diff
@@ -0,0 +1,239 @@
+TOOLS_DIR=tools
+
+# Force Go modules even when checked out inside GOPATH
+GO111MODULE := on
+export GO111MODULE
+
+REPO=github.com/apache/yunikorn-core/pkg
+# when using the -race option CGO_ENABLED is set to 1 (automatically)
+# it breaks cross compilation.
+RACE=-race
+# build commands on local os by default, uncomment for cross-compilation
+#GOOS=darwin
+#GOARCH=amd64
+
+ifeq ($(HOST_ARCH),)
+HOST_ARCH := $(shell uname -m)
+endif
+
+# Kernel (OS) Name
+OS := $(shell uname -s | tr '[:upper:]' '[:lower:]')
+
+# Allow architecture to be overwritten
+ifeq ($(HOST_ARCH),)
+HOST_ARCH := $(shell uname -m)
+endif
+
+# Build architecture settings:
+# EXEC_ARCH defines the architecture of the executables that gets compiled
+ifeq (x86_64, $(HOST_ARCH))
+EXEC_ARCH := amd64
+else ifeq (i386, $(HOST_ARCH))
+EXEC_ARCH := 386
+else ifneq (,$(filter $(HOST_ARCH), arm64 aarch64))
+EXEC_ARCH := arm64
+else ifeq (armv7l, $(HOST_ARCH))
+EXEC_ARCH := arm
+else
+$(info Unknown architecture "${HOST_ARCH}" defaulting to: amd64)
+EXEC_ARCH := amd64
+endif
+
+# shellcheck
+SHELLCHECK_VERSION=v0.9.0
+SHELLCHECK_PATH=${TOOLS_DIR}/shellcheck-$(SHELLCHECK_VERSION)
+SHELLCHECK_BIN=${SHELLCHECK_PATH}/shellcheck
+SHELLCHECK_ARCHIVE := shellcheck-$(SHELLCHECK_VERSION).$(OS).$(HOST_ARCH).tar.xz
+ifeq (darwin, $(OS))
+ifeq (arm64, $(HOST_ARCH))
+SHELLCHECK_ARCHIVE := shellcheck-$(SHELLCHECK_VERSION).$(OS).x86_64.tar.xz
+endif
+else ifeq (linux, $(OS))
+ifeq (armv7l, $(HOST_ARCH))
+SHELLCHECK_ARCHIVE := shellcheck-$(SHELLCHECK_VERSION).$(OS).armv6hf.tar.xz
+endif
+endif
+
+# golangci-lint
+GOLANGCI_LINT_VERSION=1.63.4
+GOLANGCI_LINT_PATH=$(TOOLS_DIR)/golangci-lint-v$(GOLANGCI_LINT_VERSION)
+GOLANGCI_LINT_BIN=$(GOLANGCI_LINT_PATH)/golangci-lint
+GOLANGCI_LINT_ARCHIVE=golangci-lint-$(GOLANGCI_LINT_VERSION)-$(OS)-$(EXEC_ARCH).tar.gz
+GOLANGCI_LINT_ARCHIVEBASE=golangci-lint-$(GOLANGCI_LINT_VERSION)-$(OS)-$(EXEC_ARCH)
+
+all:
+	$(MAKE) -C $(dir $(BASE_DIR)) build
+
+# Install tools
+.PHONY: tools
+tools: $(SHELLCHECK_BIN) $(GOLANGCI_LINT_BIN)
+
+# Install shellcheck
+$(SHELLCHECK_BIN):
+	@echo "installing shellcheck $(SHELLCHECK_VERSION)"
+	@mkdir -p "$(SHELLCHECK_PATH)"
+	@curl -sSfL "https://github.com/koalaman/shellcheck/releases/download/$(SHELLCHECK_VERSION)/$(SHELLCHECK_ARCHIVE)" \
+		| tar -x -J --strip-components=1 -C "$(SHELLCHECK_PATH)" "shellcheck-$(SHELLCHECK_VERSION)/shellcheck"
+
+# Install golangci-lint
+$(GOLANGCI_LINT_BIN):
+	@echo "installing golangci-lint v$(GOLANGCI_LINT_VERSION)"
+	@mkdir -p "$(GOLANGCI_LINT_PATH)"
+	@curl -sSfL "https://github.com/golangci/golangci-lint/releases/download/v$(GOLANGCI_LINT_VERSION)/$(GOLANGCI_LINT_ARCHIVE)" \
+		| tar -x -z --strip-components=1 -C "$(GOLANGCI_LINT_PATH)" "$(GOLANGCI_LINT_ARCHIVEBASE)/golangci-lint"
+
+.PHONY: lint
+# Run lint against the previous commit for PR and branch build
+# In dev setup look at all changes on top of master
+lint: $(GOLANGCI_LINT_BIN)
```

Comment:
We need a linter config file `.golangci.yml` as we want to get the same rules applied to this repo as we do for the other repos.